src/core_support.rs

```diff
@@ -13,13 +13,28 @@ use core::{fmt, str};
 use parser;
 use prelude::*;
 
+impl From<super::BytesError> for super::Error {
+    fn from(err: super::BytesError) -> Self {
+        super::Error::Bytes(err)
+    }
+}
+
 impl fmt::Debug for Uuid {
     #[inline]
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         fmt::LowerHex::fmt(self, f)
     }
 }
 
+impl fmt::Display for super::Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            super::Error::Bytes(ref err) => fmt::Display::fmt(&err, f),
+            super::Error::Parse(ref err) => fmt::Display::fmt(&err, f),
```

Comment:
since both rhs are the same we can reduce it to using a wildcard or an `|`

Reply:
not possible as LHS types in the variants are different

Reply:
it may be possible, but fine don't want to drag this PR cause of that.